_posts/news/2024-03-19-announcing-scalajs-1.16.0.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: post
+title: Announcing Scala.js 1.16.0
+category: news
+tags: [releases]
+permalink: /news/2023/01/26/announcing-scalajs-1.16.0/
+---
+
+
+We are excited to announce the release of Scala.js 1.16.0!
+
+The biggest highlight of this release is that we added a Scala.js-specific minifier.
+When combined with a general-purpose JavaScript minifier, our minifier should bring most of the code size benefits of the Google Closure Compiler when the latter cannot be enabled, notably when emitting ECMAScript modules.
+
+Scala.js 1.16.0 also upgrades by default to the standard library of Scala 2.12.19 and 2.13.13.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to [the tutorial]({{ BASE_PATH }}/tutorial/).
+
+If you need help with anything related to Scala.js, you may find our community [in `#scala-js` on Discord](https://discord.com/invite/scala) and [on Stack Overflow](https://stackoverflow.com/questions/tagged/scala.js).
+
+Bug reports can be filed [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.x, make sure to read [the release notes of Scala.js 1.0.0]({{ BASE_PATH }}/news/2020/02/25/announcing-scalajs-1.0.0/) first, as they contain a host of important information, including breaking changes.
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.15.x can be used with 1.16.0 without change.
+* It is *not* forward binary compatible with 1.15.x: libraries compiled with 1.16.0 cannot be used with 1.15.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.15.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Enhancements with compatibility concerns
+
+### Changes to the IR and linker APIs
+
+For tooling authors who directly manipulate the IR and linker APIs, there have been some breaking changes in that area.
+This is in line with our version policy for the linker APIs.
+
+The most likely changes you may hit are:
+
+* `FieldName` is now a composite of the defining `ClassName` and a `SimpleFieldName`.
+* `StoreModule` has no parameter anymore; it implicitly applies to storing the `this` value of the enclosing module class.
+
+## Enhancements
```

Comment:
Is it also worth mentioning general purpose size improvements? Even the un-minified size has dropped quite a bit:
https://github.com/scala-js/scala-js/compare/v1.15.0...v1.16.0#diff-539d15a8d5ba9a01a1724604b56cadf6851c57ec68dcf75a5073b6255d182de7L1967

Reply:
Good point. I added one more small section about that.